crates/deltalake-core/src/operations/merge.rs

```diff
@@ -1211,7 +1112,10 @@ impl std::future::IntoFuture for MergeBuilder {
             PROTOCOL.can_write_to(&this.snapshot)?;
 
             let state = this.state.unwrap_or_else(|| {
-                let session = SessionContext::new();
+                //TODO: Datafusion's Hashjoin has some memory issues. Running with all cores results in a OoM. Can be removed when upstream improvemetns are made.
+                let config = SessionConfig::new().with_target_partitions(1);
```

Comment:
@Blajda Why don't you configure a [Greedy memory pool](https://docs.rs/datafusion/latest/datafusion/execution/memory_pool/struct.GreedyMemoryPool.html) in the [RuntimeConfig](https://docs.rs/datafusion/latest/datafusion/execution/runtime_env/struct.RuntimeEnv.html)? Right now the `SessionContext` seems to be configured without a memory limit, so it's no surprise that it can OOM, right?

You can pass that in with [SessionContext::new_with_config_rt](https://docs.rs/datafusion/latest/datafusion/execution/context/struct.SessionContext.html#method.new_with_config_rt)

Reply:
I'll give this a try. Issues on the DF tracker suggest that even with a memory pool configured there are allocations that are made outside of the pool. As a start I think limiting memory to 80% of the user's system would be a good start.

Reply:
Enabled the mem pool and removed the cpu limit. Still leads to low memory system. I think this will require a follow up on how to tune this best.